tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php

```diff
@@ -82,31 +84,46 @@ public function createACustomerUsingCommand($customerReference, TableNode $table
         $groupProvider = CommonFeatureContext::getContainer()->get('prestashop.adapter.group.provider.default_groups_provider');
         $defaultGroups = $groupProvider->getGroups();
 
+        // Check if all required fields were provided
         $mandatoryFields = [
             'firstName',
             'lastName',
             'email',
-            'password',
         ];
-
         foreach ($mandatoryFields as $mandatoryField) {
             if (!array_key_exists($mandatoryField, $data)) {
                 throw new Exception(sprintf('Mandatory property %s for customer has not been provided', $mandatoryField));
             }
         }
+        if (!array_key_exists('password', $data) && empty($data['isGuest'])) {
+            throw new Exception('Password must be provided, if creating a registered customer');
```

Comment:
you must add this check in command and throw some CustomerConstraint exception if it fails. Then you are sure that its impossible to build "invalid command".
Then you add a test which expects that exception when you build the command without password and empty isGuest.

Reply:
What is the difference to the original `throw new Exception(sprintf('Mandatory property %s for customer has not been provided', $mandatoryField));`?

Reply:
These "required data" are only for test purposes.

Reply:
I did not make the fields optional in the command in the end, these data are always required to construct the command.

Reply:
oh right, the password field didn't change, it just got filled with random value in data handler :smile: